compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/diagnostics/coneDiagnosticToFirDiagnostic.kt

```diff
@@ -39,6 +40,10 @@ private fun ConeDiagnostic.toFirDiagnostic(
     }
     is ConeOperatorAmbiguityError -> FirErrors.ASSIGN_OPERATOR_AMBIGUITY.on(source, this.candidates)
     is ConeVariableExpectedError -> FirErrors.VARIABLE_EXPECTED.on(source)
+    is ConeValReassginmentError -> when (this.variable) {
+        is FirBackingFieldSymbol -> FirErrors.VAL_REASSIGNMENT_VIA_BACKING_FIELD_ERROR.on(source, this.variable)
```

Comment:
I am skipping reporting `VAL_REASSIGNMENT_VIA_BACKING_FIELD` as warning since it won't be needed for the future versions and reporting it requires passing the checker context to here. But let me know if I am missing something and you think we should still report warning when appropriate.